File: SchoolApp.Models/DataModels/Attendance.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SchoolApp.Models.DataModels
{
    [Table("Attendance")]
    public class Attendance
    {
        // Per day attendance record

        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]        
        public int AttendanceId { get; set; }       
        public DateTime Date { get; set; } = DateTime.Now;
        [Required]
        public AttendanceType Type { get; set; } = AttendanceType.Student;
        [Required]
        public int AttendanceIdentificationNumber { get; set; } = 111;
        public string? Description { get; set; }
        public bool IsPresent { get; set; } = true;          
    }

    public enum AttendanceType
    {
        Student,
        Staff
    }
}

File: SchoolAppClient.NG/src/app/Models/attendance.ts
export class Attendance {
  attendanceId!: number;
  date!: Date;
  type!: AttendanceType;
  attendanceIdentificationNumber!: number;
  description?: string;
  isPresent!: boolean;
}

export enum AttendanceType {
  Student = 0,
  Staff = 1
}
